src/schemathesis/schemas.py

```diff
@@ -398,11 +416,39 @@ def get_content_types(self, endpoint: Endpoint, response: GenericResponse) -> Li
         return self.raw_schema.get("produces", [])
 
 
-class OpenApi30(SwaggerV20):  # pylint: disable=too-many-ancestors
+@attr.s()  # pylint: disable=too-many-ancestors
+class OpenApi30(SwaggerV20):
     nullable_name = "nullable"
     example_field = "example"
     operations = SwaggerV20.operations + ("trace",)
 
+    def __attrs_post_init__(self) -> None:
```

Comment:
I'd firmly avoid double iteration on paths. The first one will happen here and another one in `get_all_endpoints` if there is no filter by `operationId` (in this case there some traversing as well). I'd say that having extra runtime cost for comparing `operationId` value with the filter in `get_all_endpoints` has a lower performance impact and much easier to implement than the currently proposed approach.

I don't think that filtration by `operationId` should be implemented differently than other filters as well as having uniqueness check at all (it may be done but I'd say in a different pull request / issue)

Reply:
Sorry, I not get it, about `double iteration`. I will read it again after done this PR, will consult you then